src/components/main-view/main-view.jsx

```diff
@@ -1,62 +1,162 @@
+  const [filteredMovies, setFilteredMovies] = useState([]);
+  const moviesURL = "https://myquickmovieapi.onrender.com/movies"
+
   useEffect(() => {
-    fetch("https://myquickmovieapi.onrender.com/movies")
+    if (!token) return;
+
+    fetch(moviesURL, {
+      headers: { Authorization: `Bearer ${token}` },
+    })
       .then((response) => response.json())
-      .then((data) => {
-        console.log(data);
-        const moviesFromApi = data.map((movie) => {
-          return {
-            _id: movie._id,
-            Title: movie.Title,
-            Description: movie.Description,
-            Genre: {
-              Name: movie.Genre.Name,
-              Description: movie.Genre.Description
-            },
-            Director: {
-              Name: movie.Director.Name,
-              Bio: movie.Director.Bio,
-              Birth: movie.Director.Birth,
-              Death: movie.Director.Death
-            },
-            ImagePath: movie.ImagePath,
-            Featured: movie.Featured
-          };
-        });
-
-        setMovies(moviesFromApi);
+      .then((data) => { 
+        setMovies(data);
       });
-  }, []);
-
- 
-
-  if (selectedMovie) {
-    return (
-      <MovieView movie={selectedMovie} onBackClick={() => setSelectedMovie(null)} />
-    );
-  }
-
-  if (movies.length === 0) {
-    return <div>The list is empty!</div>;
-  }
-    return (
-    <div>
-        {movies.map((movie) => (
-          <MovieCard
-            key={movie._id}
-            movie={movie}
-            onMovieClick={(newSelectedMovie) => {
-              setSelectedMovie(newSelectedMovie);
-            }}
+  }, [token]);
+
+  const getSearchedMovies = (arr, query) => {
+    return arr.filter((movie) => {
+      return movie.Title.toLowerCase().includes(query.toLowerCase());
+    });
+  };
+  console.log(getSearchedMovies(movies, search));
+
+  useEffect(() => {
+    setFilteredMovies(getSearchedMovies(movies, search));
+  }, [search, movies]);
+
+  return (
+    <BrowserRouter>
+      <NavigationBar
+        user={user}
+        movies={movies}
+        search={search}
+        setSearch={setSearch}
+        onLoggedOut={() => {
+          setUser(null);
+          setToken(null);
+          localStorage.clear();
+        }}
+      />
+      <Row className="justify-content-md-center">
+        <Routes>
+          <Route
+            path="/signup"
+            element={
+              <>
+                {user ? (
+                  <Navigate to="/" />
+                ) : (
+                  <Col md={5}>
+                    <SignupView />
+                  </Col>
+                )}
+              </>
+            }
           />
-      ))}
-    </div>
-    );
+          <Route
+            path="/login"
+            element={
+              <>
+                {user ? (
+                  <Navigate to="/" />
+                ) : (
+                  <Col md={5}>
+                    <LoginView onLoggedIn={(user) => setUser(user)} />
```

Comment:
After a login, the token remains null because of the prop `onLoggedIn`. The function was defined in this components with one parameter. But, the prop function was called in `login` component with two arguments. JavaScript won't consider this an error but at the execution time the second argument(`data.token` in your case) will be ignored.